nsqd/topic.go

```diff
@@ -484,11 +483,19 @@ func (t *Topic) IsPaused() bool {
 }
 
 func (t *Topic) GenerateID() MessageID {
-retry:
-	id, err := t.idFactory.NewGUID()
-	if err != nil {
+	var i int64 = 0
+	for {
+		id, err := t.idFactory.NewGUID()
+		if err == nil {
+			return id.Hex()
+		}
+		//avoid log output too many times, so just print the first error and sampling
+		if i%10000 == 0 {
+			t.ctx.nsqd.logf(LOG_ERROR,
+				"TOPIC(%s): failed to create guid - %s", t.name, err)
+		}
 		time.Sleep(time.Millisecond)
-		goto retry
+		i++
+		continue
```

Comment:
I don't think any `continue` is needed, that's what happens normally at the end of the loop body

Reply:
yes, you are right

Reply:
i delete the redundant continue